Fix tests

fd0 opened the pull request:
This PR includes a few fixes suggested in https://github.com/restic/restic/pull/259#issuecomment-131579346

Connects #259

fd0 committed:
fuse: disable for OpenBSD

fd0 committed:
Vagrantfile: Disable default rsync on /vagrant

fd0 committed:
SetupTarTestFixture: don't depend on "sh"

fd0 committed:
crypto tests: remove dependency on /dev/urandom

klauspost commented on fuse/dir.go:
```diff
@@ -1,3 +1,5 @@
+// +build !openbsd
```
Adding `// +build !windows` (as a separate line) will do the same for Windows. So while you are at it we might add it as well.

fd0 replied:
Yes, that's the plan.

klauspost commented on test/helpers.go:
```diff
@@ -88,19 +88,18 @@ func RandomReader(seed, size int) *bytes.Reader {
-// stdout/stderr.
-func System(command string, args ...string) error {
-	cmd := exec.Command(command, args...)
+	cmd := exec.Command("tar", "xzf", "-")
```
Could we use archive/tar instead?

fd0 replied:
I'm using tar files for tests in two cases:
* Providing sample repositories e.g. to test the check functions
* Provide test data to backup

The latter case includes symlinks, hard links, device nodes etc., all that must be extracted from the tar and properly set up so that the tests can run. I think it's better to rely on an external `tar` for this, at least for now.

klauspost replied:
No problem. The only issue I am seeing is that os.RemoveAll(tempdir) fails because the tar file contains readonly files, which Windows cannot delete due to https://github.com/golang/go/issues/9606

klauspost replied:
There is some workaround code here:

https://github.com/golang/go/blob/3b7841b3aff9204f054ffabbe4dd39d3e3dd3e91/src/cmd/go/go_test.go#L557

klauspost replied:
... and here it is, so it can be included in helpers.go

```Go
// resetReadOnlyFlagAll resets windows read-only flag
// set on path and any children it contains.
// The flag is set by git and has to be removed.
// os.Remove refuses to remove files with read-only flag set.
func resetReadOnlyFlagAll(t testing.TB, path string) {
	fi, err := os.Stat(path)
	if err != nil {
		t.Fatalf("resetReadOnlyFlagAll(%q) failed: %v", path, err)
	}
	if !fi.IsDir() {
		err := os.Chmod(path, 0666)
		if err != nil {
			t.Fatalf("resetReadOnlyFlagAll(%q) failed: %v", path, err)
		}
	}
	fd, err := os.Open(path)
	if err != nil {
		t.Fatalf("resetReadOnlyFlagAll(%q) failed: %v", path, err)
	}
	defer fd.Close()
	names, _ := fd.Readdirnames(-1)
	for _, name := range names {
		resetReadOnlyFlagAll(t, path+string(filepath.Separator)+name)
	}
}
```

If I insert `resetReadOnlyFlagAll(t, tempdir)` right after `	SetupTarTestFixture(t, tempdir, repoFixture)` it helps a lot.

fd0 committed:
add build constraints for windows

fd0 committed:
test helpers: add RemoveAll and ResetReadOnly

This is mainly needed in Windows, where files and dirs cannot be
removed unless they are writeable.
